File: test/fixtures/output/node/request/full.js
var request = require("request");

var jar = request.jar();
jar.setCookie(request.cookie("foo=bar"), "http://mockbin.com/har");
jar.setCookie(request.cookie("bar=baz"), "http://mockbin.com/har");

request({
  "method": "POST",
  "url": "http://mockbin.com/har",
  "qs": {
    "foo": [
      "bar",
      "baz"
    ],
    "baz": "abc",
    "key": "value"
  },
  "headers": {
    "content-type": "application/x-www-form-urlencoded",
    "accept": "application/json"
  },
  "form": {
    "foo": "bar"
  },
  "jar": jar
}, function (error, response, body) {
  if (error) throw new Error(error);

  console.log(body);
});

File: test/fixtures/output/swift/nsurlsession/full.swift
import Foundation

let headers = [
  "cookie": "foo=bar; bar=baz",
  "accept": "application/json",
  "content-type": "application/x-www-form-urlencoded"
]

var postData = NSMutableData(data: "foo=bar".dataUsingEncoding(NSUTF8StringEncoding)!)

var request = NSMutableURLRequest(URL: NSURL(string: "http://mockbin.com/har?foo=bar&foo=baz&baz=abc&key=value")!,
                                        cachePolicy: .UseProtocolCachePolicy,
                                    timeoutInterval: 10.0)
request.HTTPMethod = "POST"
request.allHTTPHeaderFields = headers
request.HTTPBody = postData

let session = NSURLSession.sharedSession()
let dataTask = session.dataTaskWithRequest(request, completionHandler: { (data, response, error) -> Void in
  if (error != nil) {
    println(error)
  } else {
    let httpResponse = response as? NSHTTPURLResponse
    println(httpResponse)
  }
})

dataTask.resume()
